ssl_cert_tracker/models.py

```diff
@@ -13,63 +13,93 @@
-    bits =  models.CharField(max_length=100, blank=True, null=True)
-    md5 =  models.CharField(max_length=100, blank=True, null=True)
-    sha1 =  models.CharField(max_length=100, blank=True, null=True)
-    
-    xml_data =  models.TextField(blank=True, null=True)
+    common_name = models.CharField(max_length=100, blank=True, null=True)
+    organization_name = models.CharField(max_length=100, blank=True, null=True)
+    country_name = models.CharField(max_length=100, blank=True, null=True)
+    sig_algo = models.CharField(max_length=100, blank=True, null=True)
+    name = models.CharField(max_length=100, blank=True, null=True)
+    bits = models.CharField(max_length=100, blank=True, null=True)
+    md5 = models.CharField(max_length=100, blank=True, null=True)
+    sha1 = models.CharField(max_length=100, blank=True, null=True)
+
+    @classmethod
+    def get_user(cls, username=settings.ORION_USER):
+        """get_user. """
+        user = cls.get_or_create_user(username)
+        return user
+
 
     def update_cert_history(self, pk_val=False):
+        """update_cert_history. """
         if pk_val:
-            orion_id_count = TestCertsHistory.objects.filter(orion_id = self.orion_id).count()
-            print(orion_id_count)
-            if orion_id_count == 0:
-                print ("We are doing insert....")
-                obj = TestCertsHistory(cert_id = int(pk_val), orion_id=self.orion_id, md5=self.md5, status="new",xml_data=self.xml_data )
+            orion_id_count = NmapHistory.objects.filter(orion_id=self.orion_id).count()
+            if orion_id_count == 0: #this is insert
+                obj = NmapHistory(cert_id=int(pk_val), \
+                orion_id=self.orion_id, \
+                md5=self.md5, \
+                status="new", \
+                xml_data=self.xml_data)
                 obj.save()
             else: # this is either update/insert
-                md5_count = TestCertsHistory.objects.filter(orion_id=self.orion_id).filter(md5=self.md5).count()
+                md5_count = NmapHistory.objects.filter(orion_id=self.orion_id).filter(md5=self.md5).count()
                 if md5_count == 0:
-                    print ("We are doing insert....")
-                    obj = TestCertsHistory(cert_id=int(pk_val), \
+                    obj = NmapHistory(cert_id=int(pk_val), \
                                            orion_id=self.orion_id, \
                                            md5=self.md5, \
                                            status="changed", \
                                            xml_data=self.xml_data)
                     obj.save()
                 else:
-                    print ("Certificate found.....We are doing update....")
-                    TestCertsHistory.objects.filter(orion_id=self.orion_id).update(status="found")
+                    NmapHistory.objects.filter(orion_id=self.orion_id).update(status="found")
 
     def save(self, *args, **kwargs):
```

Comment:
if tbe save(0 method fails, the update_cert_history() method is never called. this may result in unrecoverable transaction errors. tge update_cert_history() needs to be invoked in the post_save signal of this model